datafusion/common/src/tree_node.rs

```diff
@@ -128,17 +85,10 @@ pub trait TreeNode: Sized {
         &self,
         visitor: &mut V,
     ) -> Result<TreeNodeRecursion> {
-        match visitor.f_down(self)? {
-            TreeNodeRecursion::Continue => {
-                handle_visit_recursion!(
-                    self.apply_children(&mut |n| n.visit(visitor))?,
-                    UP
-                );
-                visitor.f_up(self)
-            }
-            TreeNodeRecursion::Jump => visitor.f_up(self),
-            TreeNodeRecursion::Stop => Ok(TreeNodeRecursion::Stop),
-        }
+        visitor
```

Comment:
❤️  -- this is really nice. I would have helped when i was fooling around with the TreeNodeMutator